torchgen/api/cpp.py

```diff
@@ -64,7 +64,9 @@
 
 def name(func: FunctionSchema, *, faithful_name_for_out_overloads: bool = False) -> str:
     name = str(func.name.name)
-    if func.is_out_fn():
+    if 'functional' in func.name.overload_name:
```

Comment:
now that these are all codegen'd, it's unlikely that anyone will call into the C++ API's for these functional ops directly. As Ed pointed out in prior feedback though, this still feels like the better thing to do, since otherwise we'll end up with two C++ API's that are identical in every way except for const-ness of their tensor arguments, which is super subtle / makes it easy to accidentally call the wrong op.